### PR TITLE
Add ColorScheme and theming to Prism

### PR DESCRIPTION
**Pull request by chriszarate**
Adapt `ColorScheme` for Prism and use a theming addon to make it possible to preview components in different color schemes! 🎉 

color-scheme

The behavior of `ColorScheme` has been modified to accept one scheme at a time. This makes it a lot easier to understand and use, but means that for most use cases you need to render multiple `ColorScheme`s, not just one. Which kinda makes sense!

Note that we are not theming Storybook itself, which will remain in "light" mode. This tracks how other addons like `viewport` work: You use them to alter the Canvas, but they don't affect Storybook itself.

### CODE REVIEW

File: .storybook/addons/StoryTheme/StoryTheme.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+				type="default"
+			/>
+			{children}
+		</>
+	);
+}
+
+StoryTheme.propTypes = {
+	children: PropTypes.node,
+	themeName: PropTypes.string.isRequired,
```
Comment (k-vosswinkel) [inline .storybook/addons/StoryTheme/StoryTheme.jsx:40]:
```suggestion
	themeName: PropTypes.oneOf([ 'At Work Dark', 'At Work Light', 'Dark', 'Light' ]).isRequired,
```

File: .storybook/addons/StoryTheme/StoryTheme.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ColorScheme, { schemes } from '../../../src/components/ColorScheme/ColorScheme';
+
+/*
+	A decorator component that provides baseline conditions required to
+	display library components. All story components are automatically
+	wrapped (see .storybook/preview.js)
+
+	https://storybook.js.org/docs/addons/introduction/#1-decorators
```
Comment (k-vosswinkel) [inline .storybook/addons/StoryTheme/StoryTheme.jsx:10]: This just lands on the default addons page and there's no mention of decorators. Maybe this instead? 
```suggestion
	https://storybook.js.org/docs/react/writing-stories/decorators
```

File: src/components/ColorScheme/ColorScheme.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Some colors are optional; note the described fallback behavior in the prop
+descriptions. Other color values are not configurable and are calculated
+automatically from base values.
+
+## Usage guidelines
+
+- **Do** supply colors in hexadecimal notation, e.g., `#000000`.
+- **Do** define official color schemes inside Prism alongside the existing `schemes`.
+- **Do** use colors from `@quartz/styles` when defining color schemes, modifying it if needed.
+- **Do** wrap `ColorScheme` in `Helmet` to hoist it to the `<head>`.
```
Comment (k-vosswinkel) [inline src/components/ColorScheme/ColorScheme.stories.mdx:41]: Is this qz-react specific?

File: .storybook/addons/StoryTheme/StoryTheme.jsx (hunk 1 above)
Comment (k-vosswinkel) [reply .storybook/addons/StoryTheme/StoryTheme.jsx:40]: Or ...maybe I'm over-thinking this, now, but what if we passed in `'WORK_LIGHT'`, `'WORK_DARK'`, `'LIGHT'`, or `'DARK'` with a default values of `'LIGHT'` and then we could use `schemes[themeName]`?

File: .storybook/addons/StoryTheme/StoryTheme.jsx (hunk 2 above)
Comment (chriszarate) [reply .storybook/addons/StoryTheme/StoryTheme.jsx]: ah yeah, this was bad copy/paste anyway

File: src/components/ColorScheme/ColorScheme.stories.mdx (hunk 3 above)
Comment (chriszarate) [reply src/components/ColorScheme/ColorScheme.stories.mdx:41]: well ... all apps should follow the advice to hoist it into the head. they don't have to use `react-helmet`, but ... it's kind of the only game in town. i'll clarify a tiny bit

File: .storybook/addons/StoryTheme/StoryTheme.jsx (hunk 1 above)
Comment (chriszarate) [reply .storybook/addons/StoryTheme/StoryTheme.jsx]: unfortunately that's not how the storybook addon works. we get the "display name" of the theme shown in the drop down

File: src/components/ColorScheme/ColorScheme.stories.mdx (hunk 3 above)
Comment (k-vosswinkel) [reply src/components/ColorScheme/ColorScheme.stories.mdx]: Awesome - sounds good. I know next to nothing about helmet, so this was more for my own education 😂 😬

**Pull request merged**